src/imageio/format/png.c

```diff
@@ -226,12 +226,21 @@ int write_image(dt_imageio_module_data_t *p_tmp, const char *filename, const voi
   {
     /* swap bytes of 16 bit files to most significant bit first */
     png_set_swap(png_ptr);
-
-    for(unsigned i = 0; i < height; i++) row_pointers[i] = (png_bytep)((uint16_t *)ivoid + (size_t)4 * i * width);
+#ifdef _OPENMP
+#pragma omp parallel for simd default(none) \
+  dt_omp_firstprivate(row_pointers, width, height, ivoid) \
+  schedule(simd:static)
+#endif
+    for(size_t i = 0; i < height; ++i) row_pointers[i] = (png_bytep)((uint16_t *)ivoid + i * width * 4);
```

Comment:
Have you benchmarked this?  It's probably slower to fire off multiple threads than compute this relatively small number of offsets single-threaded....

Reply:
I have not indeed, and wondered as well if it was worth it. I'll drop the PNG changes, thanks for the feedback.